src/main/kotlin/utils/Utils.kt

```diff
@@ -25,4 +25,13 @@ object Utils {
             action?.run()
         } while (action != null)
     }
+
+    private const val defaultCacheRevision = 209
```

Comment:
This isn't very sustainable. I played with the idea of having a revision inputbox on the CacheChooserController but 1) not the greatest UX, 2) need to wrap the CacheLibrary object with a new class to add revision field and pass it through - big refactor

Reply:
I guess it could be as simple as a checkbox (like "cache is version 209 or newer") but otherwise I think the regex approach is fine. I think the regex provided is a little over-the-top (going all-in on day-range and month-range validation when that is not the main focus of the regex) but otherwise it seems like a good idea for a machine-provided guess.

If the worst comes to the worst, we could start up a temporary region loader and ask it to load some fairly well-known area like Lumbridge and see if it gets a bunch of 0-values where it shouldn't. Not sure exactly what that would look like but I'm sure the repeating pattern of obviously incorrect values can be somehow made as detectable to a computer as it is to the human eye in the loaded cache.

> need to wrap the CacheLibrary object with a new class to add revision field and pass it through - big refactor

I'm down.

Reply:
Done